flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java

```diff
@@ -0,0 +1,141 @@
+						link("../dev/task_failure_recovery.html#failure-rate-restart-strategy", "here")),
+					text(
+						"%s: Fully qualified name of %s factory which has has a method %s.",
+						code("org.foobar.MyRestartStrategyFactoryFactory"),
+						code("RestartStrategyFactory"),
+						code("RestartStrategyFactory createFactory(Configuration configuration)"))
+				)
+				.text(
+					"If checkpointing is disabled, the default value is %s. " +
+						"If checkpointing is enabled, the default value is %s with %s restart attempts and '%s' delay.",
+					code("none"),
+					code("fixed-delay"),
+					code("Integer.MAX_VALUE"),
+					code("0 s"))
+				.build());
+
+	public static final ConfigOption<Integer> RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS = ConfigOptions
+		.key("restart-strategy.fixed-delay.attempts")
+		.defaultValue(1)
+		.withDescription(
+			Description.builder()
+				.text(
+					"The number of times that Flink retries the execution before the job is declared as failed if %s has been set to %s.",
+					code(RESTART_STRATEGY.key()),
+					code("fixed-delay"))
+				.build());
+
+	public static final ConfigOption<String> RESTART_STRATEGY_FIXED_DELAY_DELAY = ConfigOptions
+		.key("restart-strategy.fixed-delay.delay")
+		.defaultValue("0 s")
+		.withDescription(
+			Description.builder()
+				.text(
+					"Delay between two consecutive restart attempts if %s has been set to %s. " +
+						"Delaying the retries can be helpful when the program interacts with external systems where " +
+						"for example connections or pending transactions should reach a timeout before re-execution " +
+						"is attempted. It can be specified using Scala's %s notation: \"1 min\", \"20 s\"",
+					code(RESTART_STRATEGY.key()),
+					code("fixed-delay"),
+					code("FiniteDuration"))
+				.build());
+
+	public static final ConfigOption<Integer> RESTART_STRATEGY_FAILURE_RATE_MAX_FAILURES_PER_INTERVAL = ConfigOptions
+		.key("restart-strategy.failure-rate.max-failures-per-interval")
+		.defaultValue(1)
+		.withDescription(
+			Description.builder()
+				.text(
+					"Maximum number of restarts in given time interval before failing a job if %s has been set to %s.",
+					code(RESTART_STRATEGY.key()),
+					code("failure-rate"))
+				.build());
+
+	public static final ConfigOption<String> RESTART_STRATEGY_FAILURE_RATE_FAILURE_RATE_INTERVAL = ConfigOptions
+		.key("restart-strategy.failure-rate.failure-rate-interval")
+		.defaultValue("1 min")
+		.withDescription(
+			Description.builder()
+				.text(
+					"Time interval for measuring failure rate if %s has been set to %s. " +
+						"It can be specified using Scala's %s notation: \"1 min\", \"20 s\"",
+					code(RESTART_STRATEGY.key()),
+					code("failure-rate"),
+					code("FiniteDuration"))
+				.build());
+
+	public static final ConfigOption<String> RESTART_STRATEGY_FAILURE_RATE_DELAY = ConfigOptions
+		.key("restart-strategy.failure-rate.delay")
+		.defaultValue(AkkaOptions.ASK_TIMEOUT.defaultValue())
```

Comment:
Not sure why whether it's a good choice to use AKKA ask timeout as the default value? And seems there is no logic to use this default value.

I'm thinking whether we can simplify various restart delay as well. Currently AkkaOptions.WATCH_HEARTBEAT_INTERVAL(for failure rate strategy) and  AkkaOptions.WATCH_HEARTBEAT_PAUSE(for deprecated configs, already simplified in this PR) are used as default delay in some cases, which also makes it hard to describe the behavior.

Reply:
Please take a look at #9552 where this has been corrected. The intention of this PR is to simply migrate the existing options.